### PR TITLE
feat: Konnect custom DCR docs

### PR DESCRIPTION
**Pull request by c3pko**
### Description

Konnect will now allow customers to create a custom implementation for DCR so they can use an IdP of their choice. We need to point them to the example implementation so they know how to configure it, explain how the configuration works, and also show them the API specs they can use for automation.
 

DOCU-3554
### Testing instructions

Preview link: https://deploy-preview-6662--kongdocs.netlify.app/konnect/dev-portal/applications/dynamic-client-registration/custom/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+{{site.konnect_short_name}} Dev Portal supports a variety of the most widely adopted identity provider (IdP) for [Dynamic Client Registration (DCR)](/konnect/dev-portal/applications/dynamic-client-registration/):
+
+* Auth0
+* Azure
+* Curity
+* Okta
+
+If your third-party IdP is not on this list, you can still use your IdP with {{site.konnect_short_name}} by using Kong’s open source HTTP DCR bridge. This HTTP DCR bridge acts as a proxy and translation layer between your IdP and DCR applications in the Dev Portal. This bridge allows you to enable a DCR integration between the Dev Portal and your third-party IdP or IdP that isn't exposed to the internet. This integration can act as either a translation layer to bridge to your network <!--either translation layer or....???-->.
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:15]: @c3pko "This integration can act as either a translation layer to bridge to your network"

With the "either" in there, I think this sentence was meant to continue? What else can the integration act as other than the translation layer?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    participant Common domain cookie
+    Developer->>Konnect Dev Portal: Create application
+    Konnect Dev Portal->>Translation layer: Create application
+    Translation layer->>Common domain cookie: POST fidm.oidc.createRP
+    Common domain cookie--)Translation layer: 200 OK and credentials
+    Translation layer->>Konnect Dev Portal: Create app (with credentials from CDC)
+    Konnect Dev Portal->>Developer: Show credentials
+{% endmermaid %}
+
+> _**Figure 1:** This diagram shows how Kong's HTTP DCR bridge generates credentials from an IdP when a developer creates and application. First, a developer tries to create an application in {{site.konnect_short_name}} Dev Portal. This triggers Dev Portal to pass the information to the translation layer, which then sends a `POST fidm.oidc.createRP` request to the common domain cookie (CDC). Once the CDC receives the request, it sends a `200` status code in return, if successful, as well as credentials for the developer. The translation layer then creates the app with the credentials from the CDC, and then the Dev Portal shows the credentials to the developer._
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:31]: @c3pko Just to clarify that I'm understanding how this works, how is the developer "shown" the credentials? Is the CDC part of the IdP? I'm not sure where the IdP fits in the diagram.

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Configure custom DCR using the {{site.konnect_short_name}} Dev Portal DCR Handler
+
+To use this HTTP DCR bridge, you must integrate the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml)<!--integrate with what? replace this link with a link to the spec in our docs-->. Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation.
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]: @c3pko What do you integrate the DCR handler with? Your IdP? Konnect?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Configure custom DCR using the {{site.konnect_short_name}} Dev Portal DCR Handler
+
+To use this HTTP DCR bridge, you must integrate the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml)<!--integrate with what? replace this link with a link to the spec in our docs-->. Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation. 
+
+Any operation that does not return a `2xx` status code is considered failing and won't continue the process of trying to create an application in your {{site.konnect_short_name}} portal.
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]: @c3pko Does "operation" here mean what the CDC passes to the translation layer?

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md (hunk 1 above)
Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:15]:
```suggestion
If your third-party IdP is not on this list, you can still use your IdP with {{site.konnect_short_name}} by using a custom HTTP DCR bridge. This HTTP DCR bridge acts as a proxy and translation layer between your IdP and DCR applications in the Dev Portal. When a developer creates a DCR application in the Dev Portal, Konnect calls your HTTP DCR bridge which can translate the application data into a format suitable for your third-party IdP. 
```

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md (hunk 3 above)
Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]:
```suggestion
To use an unsupported IdP with DCR, you must implement the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml). Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example HTTP DCR bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation. 
```

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md (hunk 4 above)
Comment (nateslo) [reply app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]: Operation means an HTTP request in this context

Comment (nateslo) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:37]:
```suggestion
Any request that does not return a `2xx` status code is considered failing and won't continue the process of trying to create an application in your {{site.konnect_short_name}} portal.
```

**Comment by cloudjumpercat** [comment]
@c3pko @nateslo Thanks for the feedback! I just did some of my own copyedits, so I think the only thing I have questions about is if the custom DCR spec will be a new spec or an addition to an existing spec? This will help me know what additional things I should add to this PR depending on if it's new or just an addition to existing.

**Comment by nateslo** [comment]
@cloudjumpercat The spec currently lives in the example repo here: https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml. 

I don't think we want to host the spec in docs because we don't actually have an API that backs the spec. Customers need to implement an API to conforms to this spec to use the HTTP DCR feature

File: app/konnect/dev-portal/applications/dynamic-client-registration/custom.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+## Configure custom DCR using the {{site.konnect_short_name}} Dev Portal DCR Handler
+
+To use an unsupported IdP with DCR, you must implement the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml). Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example HTTP DCR bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation.
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/custom.md:35]:
```suggestion
To use an unsupported IdP with DCR, you must implement an API that conforms to the [{{site.konnect_short_name}} Dev Portal DCR Handler spec](https://github.com/Kong/konnect-portal-dcr-handler/blob/main/openapi/openapi.yaml). Kong provides an example reference implementation in the [{{site.konnect_short_name}} Dev Portal DCR Handler repository](https://github.com/Kong/konnect-portal-dcr-handler). This is an example HTTP DCR bridge implementation and is not meant to be deployed in production. We encourage you to use this implementation as a guide to create your own implementation. 
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM! Thanks for making these changes and for the feedback! This can be merged once the feature goes GA (docs will be published immediately once the PR is merged).

**Comment by nateslo** [comment]
> LGTM! Thanks for making these changes and for the feedback! This can be merged once the feature goes GA (docs will be published immediately once the PR is merged).

Thank you @cloudjumpercat!

**Pull request merged**